src/http.cc

```diff
@@ -1436,6 +1451,8 @@ HttpStateData::decodeAndWriteReplyBody()
             lastChunk = 1;
             flags.do_next_read = false;
             markParsedVirginReplyAsWhole("http parsed last-chunk");
+        } else if (eof) {
+            failOnPrematureReplyBodyEof();
```

Comment:
I know this is based on the existing FwdStateData::fail(), but that legacy method name is rather misleading. Let's rename _this_ method to avoid an implication that it actually does something like triggering a failure handling/processing code.

```suggestion
            markPrematureReplyBodyEofFailure();
```

Reply:
Ok.